lib/src/apiTest/kotlin/com/lemonappdev/konsist/architecture/optionalparameters/OptionalParametersTest.kt

```diff
@@ -0,0 +1,495 @@
+package com.lemonappdev.konsist.architecture.optionalparameters
+
+import com.lemonappdev.konsist.api.Konsist
+import com.lemonappdev.konsist.api.architecture.KoArchitectureCreator
+import com.lemonappdev.konsist.api.architecture.KoArchitectureCreator.assertArchitecture
+import com.lemonappdev.konsist.api.architecture.Layer
+import com.lemonappdev.konsist.core.exception.KoAssertionFailedException
+import com.lemonappdev.konsist.core.filesystem.PathProvider
+import io.kotest.assertions.throwables.shouldThrow
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.jupiter.api.Test
+
```

Comment:
1. I also added the `additionalMessage` parameter. Is this ok or should I remove it (the ticket only mentions the `testName` parameter)?

2. Is `OptionalParametersTest a good test class name for the above?